test/Reader/ReaderTest.php

```diff
@@ -309,6 +319,37 @@ public function testImportStringMethodThrowProperExceptionOnEmptyString()
         $feed = Reader\Reader::importString($string);
     }
 
+    public function testSetHttpFeedClient()
+    {
+        $client = $this->getMock('Zend\Feed\Reader\Http\ClientInterface');
+        Reader\Reader::setHttpClient($client);
+        $this->assertEquals($client, Reader\Reader::getHttpClient());
+    }
+
+    public function testSetHttpClient()
+    {
+        $client = $this->prophesize(ClientInterface::class)->reveal();
+        Reader\Reader::setHttpClient($client);
+        $this->assertSame($client, Reader\Reader::getHttpClient());
+    }
+
+    public function testSetHttpClientWillDecorateAZendHttpClientInstance()
+    {
+        $client = new HttpClient();
+        Reader\Reader::setHttpClient($client);
+        $cached = Reader\Reader::getHttpClient();
+        $this->assertInstanceOf(ClientInterface::class, $cached);
+        $this->assertAttributeSame($client, 'client', $cached);
+    }
+
+    /**
+     * @expectedException Zend\Feed\Reader\Exception\InvalidHttpClientException
```

Comment:
While valid I prefer the `setExpectedException` method along the PHP 5.5 `::class` (Better use traceability)